core/src/main/java/org/apache/iceberg/avro/ValueReader.java

```diff
@@ -23,4 +23,8 @@
 
 public interface ValueReader<T> {
   T read(Decoder decoder, Object reuse) throws IOException;
+
+  default void skip(Decoder decoder) throws IOException {
```

Comment:
Since skipping isn't done by `ResolvingDecoder`, this is needed. I've left out implementations for now, but we can follow up with faster skipping that doesn't construct records.

Reply:
This is nice, since this allows us to efficiently skip over the metadata when using the new block-encoder that will be part of the new Avro release: https://github.com/apache/iceberg/blob/6c344dbbbe0fc1e373f588a1bbc4c64029cc7f8c/core/src/main/java/org/apache/iceberg/avro/AvroIO.java#L166